setup/productionize/persistence/clouddriver-sql.md

```diff
@@ -0,0 +1,150 @@
+
+```yaml
+sql:
+  enabled: true
+  taskRepository:
+    enabled: true
+  cache:
+    enabled: true
+    # These parameters were determined to be optimal via benchmark comparisons
+    # in the Netflix production environment with Aurora. Setting these too low
+    # or high may negatively impact performance. These values may be sub-optimal
+    # in some environments.
+    readBatchSize: 500
+    writeBatchSize: 300
+  scheduler:
+    enabled: true
+  connectionPools:
+    default:
+      # additional connection pool parameters are available here,
+      # for more detail and to view defaults, see:
+      # https://github.com/spinnaker/kork/blob/master/kork-sql/src/main/kotlin/com/netflix/spinnaker/kork/sql/config/ConnectionPoolProperties.kt
+      default: true
+      jdbcUrl: jdbc:mysql://your.database:3306/clouddriver
+      user: clouddriver_service
+      # password: depending on db auth and how spinnaker secrets are managed
+    # The following tasks connection pool is optional. At Netflix, clouddriver
+    # instances pointed to Aurora read replicas have a tasks pool pointed at the
+    # master. Instances where the default pool is pointed to the master omit a
+    # separate tasks pool.
+    tasks:
+      user: clouddriver_service
+      jdbcUrl: jdbc:mysql://your.database:3306/clouddriver
+  migration:
+    user: clouddriver_migrate
+  jdbcUrl: jdbc:mysql://your.database:3306/clouddriver
+
+redis:
+  enabled: false
+  cache:
+    enabled: false
+  scheduler:
+    enabled: false
+  taskRepository:
+    enabled: false
+```
+
+### Agent Scheduling
+
+The above configuration parameters result in the use a table within MySQL as part of a locking service for the caching agent scheduler. This is working well for Netflix in production against Aurora, but could result in poor overall performance for all of Clouddriver depending on the database configuration. If you observe high database CPU utilization, lock contention, or if caching agents aren't consistently running at the expected frequency, you may want to try using the Redis based scheduler to determine if this is a contributing factor. The following modification to the above example configures Clouddriver to use SQL for the cache and task repository, and Redis for agent scheduling.
+
+```yaml
+sql:
+  scheduler:
+    enabled: false
+
+redis:
+  enabled: true
+  connection: redis://your.redis
+  cache:
+    enabled: false
+  scheduler:
+    enabled: true
+  taskRepository:
+    enabled: false
+```
+
+### Maintaining Task Repository Availability While Migrating from Redis to SQL
+
+If migrating from Redis to SQL in a production environment where pipeline failures or downtime should be avoided, the following modifications to the first configuration example configures Clouddriver use SQL for caching and new tasks, but with fallback reads to Redis for tasks not found in the SQL database.
```

Comment:
```suggestion
If you're migrating from Redis to SQL in a production environment, where you need to avoid pipeline failures and downtime, you can modify the configuration, as shown below, to configure Clouddriver use SQL for caching and new tasks, but with fallback reads to Redis for tasks not found in the SQL database.
```